homeassistant/components/homekit_controller/climate.py

```diff
@@ -56,15 +89,219 @@ async def async_setup_entry(hass, config_entry, async_add_entities):
 
     @callback
     def async_add_service(aid, service):
-        if service["stype"] != "thermostat":
-            return False
         info = {"aid": aid, "iid": service["iid"]}
-        async_add_entities([HomeKitClimateEntity(conn, info)], True)
-        return True
+        if service["stype"] == "thermostat":
+            async_add_entities([HomeKitClimateEntity(conn, info)], True)
+            return True
+        if service["stype"] == "heater-cooler":
+            async_add_entities([HomeKitHeaterCoolerEntity(conn, info)], True)
+            return True
+        return False
 
     conn.add_listener(async_add_service)
 
 
+class HomeKitHeaterCoolerEntity(HomeKitEntity, ClimateEntity):
+    """Representation of a Homekit climate device."""
+
+    def get_characteristic_types(self):
+        """Define the homekit characteristics the entity cares about."""
+        return [
+            CharacteristicsTypes.ACTIVE,
+            CharacteristicsTypes.CURRENT_HEATER_COOLER_STATE,
+            CharacteristicsTypes.TARGET_HEATER_COOLER_STATE,
+            CharacteristicsTypes.TEMPERATURE_COOLING_THRESHOLD,
+            CharacteristicsTypes.TEMPERATURE_HEATING_THRESHOLD,
+            CharacteristicsTypes.SWING_MODE,
+            CharacteristicsTypes.TEMPERATURE_CURRENT,
+        ]
+
+    async def async_set_temperature(self, **kwargs):
+        """Set new target temperature."""
+        temp = kwargs.get(ATTR_TEMPERATURE)
+
+        await self.async_put_characteristics(
+            {CharacteristicsTypes.TEMPERATURE_COOLING_THRESHOLD: temp}
+        )
+
+    async def async_set_hvac_mode(self, hvac_mode):
+        """Set new target operation mode."""
+        if hvac_mode == HVAC_MODE_OFF:
+            await self.async_put_characteristics(
+                {CharacteristicsTypes.ACTIVE: ActivationStateValues.INACTIVE}
+            )
+            return
+        await self.async_put_characteristics(
+            {
+                CharacteristicsTypes.TARGET_HEATER_COOLER_STATE: TARGET_HEATER_COOLER_STATE_HASS_TO_HOMEKIT[
+                    hvac_mode
+                ],
+            }
+        )
+
+    @property
+    def current_temperature(self):
+        """Return the current temperature."""
+        return self.service.value(CharacteristicsTypes.TEMPERATURE_CURRENT)
+
+    @property
+    def target_temperature(self):
+        """Return the temperature we try to reach."""
+        state = self.service.value(CharacteristicsTypes.TARGET_HEATER_COOLER_STATE)
+        if state == TargetHeaterCoolerStateValues.COOL and self.service.has(
```

Comment:
You can do something like:

```
if state == TargetHeaterCoolerStateValues.COOL:
    return self.service.value(CharacteristicsTypes.TEMPERATURE_COOLING_THRESHOLD)
```

It will return None if the char isn't present, which is what happens in your current implementation anyway.